src/supplemental/quic/msquic_dial.c

```diff
@@ -214,6 +216,33 @@ verify_peer_cert_tls(QUIC_CERTIFICATE* cert, QUIC_CERTIFICATE* chain, char *ca)
 
 }
 
+static void
+check_timeout_reopen(void *arg)
+{
+	ex_quic_conn  *ec = arg;
+	nni_quic_conn *c;
+
+	nni_mtx_lock(&ec->mtx);
+	for (int i=0; i<QUIC_SUB_STREAM_NUM; i++) {
+		if ((c = ec->substrms[i]) != NULL) {
+			nni_mtx_lock(&c->mtx);
+			if (c->tmo >= QUIC_SUB_STREAM_TIMEOUT) {
+				log_warn("[sid%d] close stream actively due to timeout%ld.", c->id, c->tmo);
+				c->tmo = 0; // reset
+				nni_mtx_unlock(&c->mtx);
+				break;
+			}
+			nni_mtx_unlock(&c->mtx);
+			c = NULL;
+		}
+	}
+	nni_mtx_unlock(&ec->mtx);
+
+	if (c)
+		quic_substream_close(c);
```

Comment:
If so, isnt it always only 1 sub stream will be closed?

Reply:
Yes. In a round. There is only one sub stream will be closed. If another sub stream also sends timeout. It will be closed at next round.